### PR TITLE
fix: Adding missing EasyMock.verify calls #1414

### LINKED ISSUE #1408: Missing EasyMock.verify in unit tests
In some unit tests, before resetting mocked objects or after mocked objects get used, `EasyMock.verify` does not get invoked on mocked objects. So, we should add it.

For example, in `UserTaskManagerTest.java` and `KafkaCruiseControlServletEndpointTest.java`. There could be more cases. So, I suggest thoroughly iterate all cases where `EasyMock` gets used.

### PR DESCRIPTION
**Pull request by yamashita-ki**
This PR resolves #<Replace-Me-With-The-Issue-Number-Addressed-By-This-PR>1408

### CODE REVIEW

File: cruise-control/src/test/java/com/linkedin/kafka/cruisecontrol/servlet/UserTaskManagerTest.java [hunk 1]
```diff
@@ -53,7 +53,7 @@ public void testCreateUserTask() {
     Assert.assertEquals(userTaskHeader.getValue(), UserTaskManager.USER_TASK_HEADER_NAME);
     Assert.assertEquals(userTaskHeaderValue.getValue(), testUserTaskId.toString());
     Assert.assertEquals(future, future1);
-
+    EasyMock.verify(mockHttpServletResponse, mockHttpSession);
```
Comment (efeg) [inline cruise-control/src/test/java/com/linkedin/kafka/cruisecontrol/servlet/UserTaskManagerTest.java:56]: Should this be `EasyMock.verify(mockUuidGenerator, mockHttpServletResponse, mockHttpSession);`?

File: cruise-control/src/test/java/com/linkedin/kafka/cruisecontrol/servlet/UserTaskManagerTest.java [hunk 2]
```diff
@@ -139,6 +139,8 @@ public void testSessionsShareUserTask() {
 
     // The 2nd request should reuse the UserTask created for the 1st request since they use the same session and send the same request.
     Assert.assertEquals(1, userTaskManager.numActiveSessionKeys());
+    EasyMock.verify(mockUuidGenerator, mockHttpServletResponse1, mockHttpServletResponse2, mockHttpServletResponse3);
+    EasyMock.verify(mockHttpServletRequest1, mockHttpServletRequest2, mockHttpServletRequest3);
```
Comment (efeg) [inline cruise-control/src/test/java/com/linkedin/kafka/cruisecontrol/servlet/UserTaskManagerTest.java:143]: Nit: Can we avoid splitting this into separate verify calls and use a single verify?

File: cruise-control/src/test/java/com/linkedin/kafka/cruisecontrol/servlet/UserTaskManagerTest.java [hunk 3]
```diff
@@ -267,7 +270,7 @@ public void testMaximumActiveTasks() {
     OperationFuture future1 =
         userTaskManager.getOrCreateUserTask(mockHttpServletRequest1, mockHttpServletResponse, uuid -> future, 0, true, null).get(0);
     Assert.assertEquals(future, future1);
-
+    EasyMock.verify(mockHttpServletRequest1, mockHttpServletResponse);
```
Comment (efeg) [inline cruise-control/src/test/java/com/linkedin/kafka/cruisecontrol/servlet/UserTaskManagerTest.java:273]: Should this be `EasyMock.verify(mockHttpSession1, mockHttpServletRequest1, mockHttpServletResponse);`?

File: cruise-control/src/test/java/com/linkedin/kafka/cruisecontrol/servlet/UserTaskManagerTest.java [hunk 4]
```diff
@@ -278,6 +281,7 @@ public void testMaximumActiveTasks() {
       OperationFuture future2 =
           userTaskManager.getOrCreateUserTask(mockHttpServletRequest2, mockHttpServletResponse, uuid -> future, 0, true, null).get(0);
       Assert.assertEquals(future, future2);
+      EasyMock.verify(mockHttpServletRequest2, mockHttpServletResponse);
```
Comment (efeg) [inline cruise-control/src/test/java/com/linkedin/kafka/cruisecontrol/servlet/UserTaskManagerTest.java:284]: Should this be `EasyMock.verify(mockHttpSession2, mockHttpServletRequest2, mockHttpServletResponse);`?

**Review by efeg — CHANGES_REQUESTED** [review]
@yamashita-ki Thanks for the PR. Left some comments

File: cruise-control/src/test/java/com/linkedin/kafka/cruisecontrol/servlet/UserTaskManagerTest.java (hunk 2 above)
Comment (yamashita-ki) [reply cruise-control/src/test/java/com/linkedin/kafka/cruisecontrol/servlet/UserTaskManagerTest.java]: If I make it simple, this Linter Error has occurred.

[ant:checkstyle] [ERROR] /home/circleci/workspace/cruise-control/src/test/java/com/linkedin/kafka/cruisecontrol/servlet/UserTaskManagerTest.java:142: Line is longer than 150 characters (found 192). Try to keep lines under 120 characters. [LineLength]

Comment (efeg) [reply cruise-control/src/test/java/com/linkedin/kafka/cruisecontrol/servlet/UserTaskManagerTest.java]: We can put a line break within a single verify call -- something like:
```
EasyMock.verify(mockUuidGenerator, mockHttpServletResponse1, mockHttpServletResponse2, mockHttpServletResponse3, 
                mockHttpServletRequest1, mockHttpServletRequest2, mockHttpServletRequest3);
```

**Pull request merged**